share/p2p/peers/manager.go

```diff
@@ -103,17 +111,21 @@ func (s *Manager) GetPeer(
 	case peerID = <-s.fullNodes.waitNext(ctx):
 		return peerID, s.doneFunc(datahash, peerID), nil
 	case <-ctx.Done():
-		return "", s.doneFunc(datahash, peerID), ctx.Err()
+		return "", nil, ctx.Err()
 	}
 }
 
 func (s *Manager) doneFunc(datahash share.DataHash, peerID peer.ID) DoneFunc {
-	return func(success bool) {
-		if success {
+	return func(result sampleResult) {
+		switch result {
+		case ResultSuccess:
 			s.markSampled(datahash)
-			return
+		case ResultFail:
+		case ResultPeerMMisbehaved:
```

Comment:
We shouldn't reject from Validator if failed to serve, we should just ignore rn (but we should figure out how this would impact connectivity / retrieval before we arrive at a decision)

Reply:
thats the point of ResultFail being a noop